app/models/df_json_wrapper.rb

```diff
@@ -4,32 +4,62 @@ def initialize(json)
     @json = json
   end
 
+  def self.json_accessor(mappings)
+    json_reader(mappings)
+    json_writer(mappings)
+  end
+
   def self.json_reader(mappings)
     mappings.each do |selector_name, selector_info|
-      selector_info => {type:, key:}
+      selector_info => { type:, key: }
+      key_path = key.split
       case type
       when :money_amount
         define_method(selector_name) do
-          df_json_value(key)&.to_d || 0
+          df_json_value(key_path)&.to_d || 0
         end
       when :date
         define_method(selector_name) do
-          Date.parse(df_json_value(key)) if df_json_value(key).present?
+          Date.parse(df_json_value(key_path)) if df_json_value(key_path).present?
         end
       when :string, :boolean
         define_method(selector_name) do
-          df_json_value(key)
+          df_json_value(key_path)
         end
       else
         define_method(selector_name) do
-          df_json_value(key)
+          df_json_value(key_path)
         end
       end
     end
   end
 
-  def df_json_value(key)
-    key_path = key.split
+  def self.json_writer(mappings)
+    mappings.each do |selector_name, selector_info|
+      selector_info => { type:, key: }
+      key_path = key.split
+      case type
+      when :date
```

Comment:
should we also handle setting a boolean value ("true") or a `money_amount` value?

Reply:
I believe but am not sure that dates are the only value whose string representation doesn't just already look the way we want, so they were the only one that needed extra formatting. But I was pretty deep into this story by the time I did this and was just trying to finish up, so maybe the others do need to be there! We don't use this for anything but strings & dates yet, in any case.